app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue

```diff
@@ -473,15 +486,37 @@ export default {
       if (isEscape(e)) {
         this.hideEmojiPicker();
         this.hideMentions();
-      } else if (isEnter(e)) {
+      } else if (!this.isCmdPlusEnterToSendMessage && isEnter(e)) {
         const hasSendOnEnterEnabled =
           (this.showRichContentEditor &&
-            this.enterToSendEnabled &&
+            !this.isCmdPlusEnterToSendMessage &&
+            !this.hasUserMention &&
+            !this.showCannedMenu) ||
+          !this.showRichContentEditor;
+        const shouldSendMessage =
+          hasSendOnEnterEnabled &&
+          !hasPressedCommand(e) &&
+          !hasPressedShift(e) &&
+          this.isFocused;
+        if (shouldSendMessage) {
+          e.preventDefault();
+          this.sendMessage();
+        }
+      } else if (
+        this.isCmdPlusEnterToSendMessage &&
+        hasPressedCommandAndEnter(e)
+      ) {
+        const hasSendOnCmdPlusEnterEnabled =
+          (this.showRichContentEditor &&
+            this.isCmdPlusEnterToSendMessage &&
             !this.hasUserMention &&
             !this.showCannedMenu) ||
           !this.showRichContentEditor;
         const shouldSendMessage =
-          hasSendOnEnterEnabled && !hasPressedShift(e) && this.isFocused;
+          hasSendOnCmdPlusEnterEnabled &&
+          hasPressedCommand(e) &&
+          !hasPressedShift(e) &&
+          this.isFocused;
         if (shouldSendMessage) {
           e.preventDefault();
```

Comment:
This logic is getting complex. Can we do the refactoring?

Reply:
@muhsin-k @pranavrajs Can you help me in refactoring? I have tried it but it's breaking.